src/dev/flang/be/jvm/JVM.java

```diff
@@ -510,28 +543,40 @@ void compile(JVM jvm, int cl)
           {
             try
               {
-                cf.write(PATH_FOR_CLASSES);
+                cf.write(jvm._jos);
               }
             catch (IOException io)
               {
                 Errors.error("JVM backend I/O error",
-                             "While creating class '" + cf.classFile() + "' in '" + PATH_FOR_CLASSES + "', received I/O error '" + io + "'");
+                             "While creating class '" + cf.classFile() + "' in JAR, received I/O error '" + io + "'");
               }
           }
         if (jvm._types.hasInterfaceFile(cl))
           {
             var ci = jvm._types.interfaceFile(cl);
             try
               {
-                ci.write(PATH_FOR_CLASSES);
+                ci.write(jvm._jos);
               }
             catch (IOException io)
               {
                 Errors.error("JVM backend I/O error",
-                             "While creating class '" + ci.classFile() + "' in '" + PATH_FOR_CLASSES + "', received I/O error '" + io + "'");
+                             "While creating class '" + ci.classFile() + "' in JAR, received I/O error '" + io + "'");
               }
           }
       }
+      void finish(JVM jvm)
+      {
+        try
+          {
+            jvm._jos.close();
```

Comment:
Would be great to also create a shell script with the base name of the main clazz that performs `java -jar <baseName>.jar`. Same should be done for `-classes` (in a speparate patch).

Reply:
No, I did not know binfmt_misc so far. Seems useful, needs to be set up by the user. I would like to provide an easy and ideally uniform way to run the created code, independent of the backend (C/jar/classes/...).